src/System.Private.CoreLib/src/System/HighPerformanceCounter.Unix.cs

```diff
@@ -8,26 +8,9 @@ namespace System
 {
     internal static class HighPerformanceCounter
     {
-        public static ulong TickCount
-        {
-            get
-            {
-                long tickCount;
-                bool success = Interop.Sys.GetTimestamp(out tickCount);
-                Debug.Assert(success);
-                return (ulong)tickCount;
-            }
-        }
+        public static ulong TickCount => Interop.Sys.GetTimestamp();
 
-        public static ulong Frequency { get; } = GetFrequency();
-
-        private static ulong GetFrequency()
-        {
-            // Cache the frequency on the managed side to avoid the cost of P/Invoke on every access to Frequency
-            long frequency;
-            bool success = Interop.Sys.GetTimestampResolution(out frequency);
-            Debug.Assert(success);
-            return (ulong)frequency;
-        }
+        // Cache the frequency on the managed side to avoid the cost of P/Invoke on every access to Frequency
+        public static ulong Frequency { get; } = Interop.Sys.GetTimestampResolution();
```

Comment:
Is there any reason why this isn't shared with CoreFX instead of having the difference mechanisms we have today?

Reply:
Do you mean to move StopWatch from CoreFX to CoreLib, and make the managed ClrThreadPool depend on it? I am not sure whether it would be an improvement.

Reply:
I was more thinking of having `Stopwatch` in CoreFX consume `HighPerformanceCounter`; rather than having private `QueryPerformanceFrequency` and `QueryPerformanceCounter` methods...

Reply:
I do not have a strong opinion. There are many places where we can share a few more lines, at the cost of increased coupling. Personally, I tend to prefer less coupling.

Reply:
Fair enough. I was thinking it might be nice in the places where there is interop code involved as they are effectively "coupled" already.

It would also ensure that minor "improvements" like the signature change from `out long` to `long*` to avoid the implicit pinning on Windows apply to both places (it looks like `HighPerformancCounter.Windows.cs` is still using `out long`).

Reply:
I also don't have a strong opinion, however, as I don't think this code will get touched that often 😄